### PR TITLE
ref(grouping): Add SDK name as a tag on event_manager metrics

### PR DESCRIPTION
**Pull request by vartec**
Adding SDK tag to event_manager metrics.

Implements: https://github.com/getsentry/sentry/issues/59053

### CODE REVIEW

File: src/sentry/event_manager.py [hunk 1]
```diff
@@ -776,7 +779,7 @@ def _auto_update_grouping(project: Project) -> None:
             "sentry:secondary_grouping_expiry": expiry,
             "sentry:grouping_config": new_grouping,
         }
-        for (key, value) in changes.items():
+        for key, value in changes.items():
```
Comment (vartec) [inline src/sentry/event_manager.py:782]: ^^ auto-formatting

**Review by lobsterkatie — COMMENTED** [review]
Technically, to match our conventions, I'd do `ref(grouping): Add SDK name...`, but otherwise, LGTM!

Can you please update the relevant tests, also? This may not find all of them, but a good place to start is to search for `mock_metrics_incr`.

File: src/sentry/event_manager.py [hunk 2]
```diff
@@ -1826,7 +1831,10 @@ def _create_group(project: Project, event: Event, **kwargs: Any) -> Group:
     except OperationalError:
         metrics.incr(
             "next_short_id.timeout",
-            tags={"platform": event.platform or "unknown"},
+            tags={
+                "platform": event.platform or "unknown",
+                "sdk": event.data.get("sdk", {}).get("name") or "unknown",
+            },
```
Comment (lobsterkatie) [inline src/sentry/event_manager.py:1837]: I don't think we need to add SDK here. There's no reason to even collect platform, since there's nothing platform-specific about incrementing an id, but getting rid of it is probably best done in a separate PR.

**Review by lobsterkatie — COMMENTED** [review]
Looking at this more closely (which I should have done the first time, sorry), a few small things:

- We use present tense rather than gerund in our PR titles/commit messages ("add" rather than "adding").
 
- There's one spot, called out below, where I don't think we need to add an SDK tag.

- I think it's worth deciding the scope of this change and either revising the PR title or pulling most of your changes into a separate PR. Of the metrics you've modified here, only the one in `_calculate_event_grouping` and the one labeled `grouping.in_app_frame_mix` are actually grouping metrics. (You'd be forgiven for thinking that the one in `_calculate_span_grouping` is also about grouping, but when we say "grouping," what we really mean is specifically event grouping.) So I'd either a) pull "grouping" out of the PR title, and make it just about `event_manager` metrics, or b) restrict it just to grouping metrics.

- In either case, I'd also add the tags in the following spots:
    - The `event_manager.apply_server_fingerprinting` metric.
    - The `event_manager.background_grouping` metric. For this one, you'll need to switch from the decorator to a context manager inside the function calling `metrics.timer` (which is what `metrics.wraps` does internaly). 
    - The `metric_tags` passed to `save_error_events` inside of `save`.

**Review by lobsterkatie — APPROVED** [review]
Good catch on the `save_event.calculate_event_grouping` metric!

**Comment by lobsterkatie** [comment]
We reverted this because it was blowing up the number of tags we're keeping track of in DD. We're going to look into doing some normalization to reduce the number of values.
